protocol/contracts/v2/circuitBreaker/TempleCircuitBreakerProxy.sol

```diff
@@ -0,0 +1,99 @@
+pragma solidity ^0.8.17;
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Temple (v2/circuitBreaker/ITempleCircuitBreakerProxy.sol)
+
+import { TempleElevatedAccess } from "contracts/v2/access/TempleElevatedAccess.sol";
+import { EnumerableSet } from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import { CommonEventsAndErrors } from "contracts/common/CommonEventsAndErrors.sol";
+import { ITempleCircuitBreaker } from "contracts/interfaces/v2/circuitBreaker/ITempleCircuitBreaker.sol";
+import { ITempleCircuitBreakerProxy } from "contracts/interfaces/v2/circuitBreaker/ITempleCircuitBreakerProxy.sol";
+
+/**
+ * @title Temple Circuit Breaker Proxy
+ * 
+ * @notice Direct circuit breaker requests to the correct underlying implementation,
+ * based on a pre-defined bytes32 identifier, and a token.
```

Comment:
Overall, I would have imagined a proxy+factory implementation which creates circuit breakers and also keeps track of them (using OZ Clones in addition). implementation can be changed by admin.

Reply:
Perhaps a factory could make sense. However I don't know how many we'll actually create and if it's warranted.

For example the plan is for the TLC breaker to be *shared* with other user facing breakers - eg when we have a Sice Bazaar amm, it'll use the same one.

Further, we may have other different types of breakers. Eg a whitelist, or user specific volumes, or whatever

So doing this externally and then just registering made most sense to me for that flexibility.

Reply:
then you can keep it as is, since you don't know how many will be created/variations.